### ISSUE TITLE
[explicit-module-boundary-types] expects return type on methods of abstract class setter

### ISSUE THREAD

**Issue by ljosberinn** [typescript-eslint/typescript-eslint#2407]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "error"
  }
}
```

```TS
export abstract class SomeClass<T> {
  abstract set value(element: T);
}
```

![image](https://user-images.githubusercontent.com/29307652/90766876-8b924680-e2ec-11ea-9d1f-ef51314bb2f7.png)

https://github.com/ljosberinn/explicit-module-boundary-types-bug

**Expected Result**
The rule should never expect a return type of a setter to be declared.

**Actual Result**

The rule expects a return type which is a compiler error.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.1` |
| `@typescript-eslint/parser`        | `3.9.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `---` |

**Comment by bradzacher** [comment]
The bug is in this code right here:
https://github.com/typescript-eslint/typescript-eslint/blob/44e4d9b1af3a7cf3bed0009686737c7f8a279596/packages/eslint-plugin/src/rules/explicit-module-boundary-types.ts#L424-L438

Should be a pretty simple fix, if you want to submit a PR.
